Source/Scene/Cesium3DTile.js

```diff
@@ -0,0 +1,1018 @@
+            var boundingSphere = tile._contentBoundingVolume.boundingSphere;
+            var sphere = BoundingSphere.projectTo2D(boundingSphere, frameState.mapProjection, scratchProjectedBoundingSphere);
+            tile._contentBoundingVolume2D = new TileBoundingSphere(sphere.center, sphere.radius);
+        }
+        return frameState.mode !== SceneMode.SCENE3D ? tile._contentBoundingVolume2D : tile._contentBoundingVolume;
+    }
+
+    /**
+     * Determines whether the tile's bounding volume intersects the culling volume.
+     *
+     * @param {FrameState} frameState The frame state.
+     * @param {Number} parentVisibilityPlaneMask The parent's plane mask to speed up the visibility check.
+     * @returns {Number} A plane mask as described above in {@link CullingVolume#computeVisibilityWithPlaneMask}.
+     *
+     * @private
+     */
+    Cesium3DTile.prototype.visibility = function(frameState, parentVisibilityPlaneMask) {
+        var cullingVolume = frameState.cullingVolume;
+        var boundingVolume = getBoundingVolume(this, frameState);
+        return cullingVolume.computeVisibilityWithPlaneMask(boundingVolume, parentVisibilityPlaneMask);
+    };
+
+    /**
+     * Assuming the tile's bounding volume intersects the culling volume, determines
+     * whether the tile's content's bounding volume intersects the culling volume.
+     *
+     * @param {FrameState} frameState The frame state.
+     * @returns {Intersect} The result of the intersection: the tile's content is completely outside, completely inside, or intersecting the culling volume.
+     *
+     * @private
+     */
+    Cesium3DTile.prototype.contentVisibility = function(frameState) {
+        // Assumes the tile's bounding volume intersects the culling volume already, so
+        // just return Intersect.INSIDE if there is no content bounding volume.
+        if (!defined(this._contentBoundingVolume)) {
+            return Intersect.INSIDE;
+        }
+
+        // PERFORMANCE_IDEA: is it possible to burn less CPU on this test since we know the
+        // tile's (not the content's) bounding volume intersects the culling volume?
+        var cullingVolume = frameState.cullingVolume;
+        var boundingVolume = getContentBoundingVolume(this, frameState);
+        return cullingVolume.computeVisibility(boundingVolume);
+    };
+
+    /**
+     * Computes the (potentially approximate) distance from the closest point of the tile's bounding volume to the camera.
+     *
+     * @param {FrameState} frameState The frame state.
+     * @returns {Number} The distance, in meters, or zero if the camera is inside the bounding volume.
+     *
+     * @private
+     */
+    Cesium3DTile.prototype.distanceToTile = function(frameState) {
+        var boundingVolume = getBoundingVolume(this, frameState);
+        return boundingVolume.distanceToCamera(frameState);
+    };
+
+    var scratchCartesian = new Cartesian3();
```

Comment:
Either move this to the top of the file or name it something more specific, like `distanceToTileCenterScratch` so it's not accidentally reused in other functions.